lib/query/query_supporters.rb

```diff
@@ -632,9 +636,18 @@ def self.for_crm_profile(npo_id, ids)
         ["donations", "donations.supporter_id=supporters.id"],
         ["full_contact_infos", "full_contact_infos.supporter_id=supporters.id"],
         ["recurring_donations", "recurring_donations.donation_id=donations.id"],
+        ["supporter_addresses", "supporters.primary_address_id = supporter_addresses.id"],
         [QuerySupporters.profile_payments_subquery, "payments.supporter_id=supporters.id"])
-      .group_by("supporters.id")
-      .where("supporters.id IN ($ids)", ids: ids)
+      .group_by(
+        "supporters.id",
+        "supporter_addresses.id",
+        "supporter_addresses.supporter_id",
+        "supporter_addresses.address",
+        "supporter_addresses.state_code",
+        "supporter_addresses.city",
+        "supporter_addresses.zip_code",
+        "supporter_addresses.country"
```

Comment:
I have the same question on the group_by here. What's the goal of this?

Reply:
The same case as I mentioned on the other comment.

Reply:
This is the resulting query for this:

```sql
SELECT
  supporters.id,
  supporters.name,
  supporters.email,
  supporter_addresses.address,
  supporter_addresses.state_code,
  supporter_addresses.city,
  supporter_addresses.zip_code,
  supporter_addresses.country,
  supporters.organization,
  supporters.phone,
  supporters.created_at,
  supporters.imported_at,
  supporters.anonymous AS anon,
  supporters.is_unsubscribed_from_emails,
  COALESCE(MAX(payments.sum), 0) AS raised,
  COALESCE(MAX(payments.count), 0) AS payments_count,
  COALESCE(COUNT(recurring_donations.active), 0) AS recurring_donations_count,
  MAX(full_contact_infos.full_name) AS fc_full_name,
  MAX(full_contact_infos.age) AS fc_age,
  MAX(full_contact_infos.location_general) AS fc_location_general,
  MAX(full_contact_infos.websites) AS fc_websites
FROM
  supporters
  LEFT JOIN donations ON donations.supporter_id = supporters.id
  LEFT JOIN full_contact_infos ON full_contact_infos.supporter_id = supporters.id
  LEFT JOIN recurring_donations ON recurring_donations.donation_id = donations.id
  LEFT JOIN supporter_addresses ON supporters.primary_address_id = supporter_addresses.id
  LEFT JOIN (
    SELECT
      supporter_id,
      SUM(gross_amount),
      COUNT(id) AS count
    FROM
      payments
    GROUP BY
      supporter_id
  ) AS "payments" ON payments.supporter_id = supporters.id
WHERE
  (supporters.id IN (3160024))
  AND (supporters.nonprofit_id = 3624)
GROUP BY
  supporters.id,
  supporter_addresses.id,
  supporter_addresses.supporter_id,
  supporter_addresses.address,
  supporter_addresses.state_code,
  supporter_addresses.city,
  supporter_addresses.zip_code,
  supporter_addresses.country
```

Reply:
That is indeed a query.